backend/battle/forms.py

```diff
@@ -49,23 +40,18 @@ def clean(self):
         if 'opponent' not in cleaned_data:
             raise forms.ValidationError("ERROR: You need to choose an opponent")
 
-        if cleaned_data['opponent'] == cleaned_data['creator']:
+        valid_creator_field = validate_if_creator_and_opponent_are_different(
+            cleaned_data['opponent'], cleaned_data['creator'])
+
+        if not valid_creator_field:
             raise forms.ValidationError("ERROR: You can't challenge yourself.")
         return cleaned_data
 
     def save(self, commit=True):
         cleaned_data = self.clean()
         instance = super().save()
         opponent = cleaned_data["opponent"]
-        if opponent.is_guest:
-            invite_form = PasswordResetForm(data={"email": opponent.email})
-            invite_form.is_valid()
-            invite_form.save(
-                self, subject_template_name='registration/guest_email_subject.txt',
-                email_template_name='registration/guest_email.html',
-                from_email=settings.FROM_EMAIL,)
-        else:
-            send_invite_email(instance.opponent, instance.creator)
+        create_battle(self, opponent, instance.creator)
```

Comment:
This appears to be only creating the opponent and sending the invite

Reply:
You're right, this only sends the email invite or the email create password, I'm going to change this :)